From 93db038ab4c4a8119d23f824aeeeb99796be8247 Mon Sep 17 00:00:00 2001
From: Oliver Neukum <oneukum@suse.com>
Date: Mon, 13 Jun 2016 14:23:04 +0200
Subject: [PATCH] usb: quirk to stop runtime PM for Intel 7260
References: bnc#984446
Patch-Mainline: Never (upstream fixed this in firmware which we can't take)

The old firmware in the Intel 7260 BT modules is prone to crashes
if runtime power management is used. Our driver is too old to use
the newest firmware. Hence runtime PM is to be disabled for this
module.

Signed-off-by: Oliver Neukum <oneukum@suse.com>
---
 drivers/usb/core/quirks.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/usb/core/quirks.c b/drivers/usb/core/quirks.c
index 6dc810b..3a0f9b6 100644
--- a/drivers/usb/core/quirks.c
+++ b/drivers/usb/core/quirks.c
@@ -208,6 +208,9 @@ static const struct usb_device_id usb_quirk_list[] = {
 	/* Blackmagic Design UltraStudio SDI */
 	{ USB_DEVICE(0x1edb, 0xbd4f), .driver_info = USB_QUIRK_NO_LPM },
 
+	/* Intel 7260 BT - with the old firmware, but we cannot update */
+	{ USB_DEVICE(0x8087, 0x07dc), .driver_info = USB_QUIRK_RESET_RESUME },
+
 	{ }  /* terminating entry must be last */
 };
 
-- 
2.1.4

